### ISSUE TITLE
Install timed out but app deployed and can't upgrade

### ISSUE THREAD

**Issue by brianharwell** [helm/helm#8263]
Output of `helm version`: 
```
version.BuildInfo{Version:"v3.2.2", GitCommit:"a6ea66349ae3015618da4f547677a14b9ecc09b3", GitTreeState:"clean", GoVersion:"go1.13.12"}
```

Output of `kubectl version`: 
```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.0", GitCommit:"9e991415386e4cf155a24b1da15becaa390438d8", GitTreeState:"clean", BuildDate:"2020-03-25T14:58:59Z", GoVersion:"go1.13.8", Compiler:"
gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.5", GitCommit:"e0fccafd69541e3750d460ba0f9743b90336f24f", GitTreeState:"clean", BuildDate:"2020-04-16T11:35:47Z", GoVersion:"go1.13.9", Compiler:"
gc", Platform:"linux/amd64"}
```
Cloud Provider/Platform (AKS, GKE, Minikube etc.): Kubernbetes in Rancher on vSphere

I am deploying a Windows container and since this is the first time I am deploying to the cluster the full image needs to download which takes a while. I ran the following command...
```
helm upgrade toggleservice .\ToggleService-1.0.30.tgz --install --wait --description "Manual deploy"
```
The result was...
```
Release "toggleservice" does not exist. Installing it now.
Error: timed out waiting for the condition
```
So after a while the image downloaded and the pods started fine.

When I get the helm history I see this...
```
REVISION        UPDATED                         STATUS  CHART                   APP VERSION     DESCRIPTION
1               Sun Jun  7 09:44:20 2020        failed  ToggleService-1.0.30    1.0.133         Release "toggleservice" failed: timed out waiting for the condition
```
That is expected but when I try deploying using the command from above I get the error:
```
Error: UPGRADE FAILED: "toggleservice" has no deployed releases
```
So I thought maybe I need to use install but when I do I get the error...
```
Error: cannot re-use a name that is still in use
```
So I can't install nor can I upgrade. I need to delete and then redeploy.

What I want to happen is if I run upgrade it works.

**Comment by marckhouzam** [comment]
I believe this has been fixed in #7653 which was released in 3.2.1.
However, I can't find it in the 3.2.2 release.  I will follow-up on this.

In the mean time, you may be interested in trying with 3.2.1 to see if it works as you need.

**Comment by mattfarina** [comment]
We seem to have an issue. Change from v3.2.1 have been dropped from the v3.2.2 release. I'm investigating. Expect a v3.2.3 release this week with fixes. I have found other things missing as well.

**Comment by mattfarina** [comment]
Please try release v3.2.3 and let us know if the issue persists.

**Comment by colearendt** [comment]
Is it possible to clarify whether the fix should
- help us get out of this situation once we are in it (from a different helm version)
- help avoid the issue happening again?

I have gotten into a very similar state as well (using `helm 3.1.2`), and `3.2.3` did not _fix_ it for me (I am still in this state). However, I suspect that deleting the secrets associated with the release will help me resolve the issue.

To clarify, the way I got into this state was by hitting `CTRL+C` during the initial deploy of a chart. I have a secret for the release but no actual release `helm list --all-namespaces` does not return this release.

**Comment by bacongobbler** [comment]
> To clarify, the way I got into this state was by hitting `CTRL+C` during the initial deploy of a chart.

> Maybe this is a different issue?

Yeah this may seem like a separate issue. Can you please open a new ticket? If you still have the output of `helm history` available that would be helpful to determing if #7653 can solve this for you.

> Is it possible to clarify whether the fix should
> 
>     * help us get out of this situation once we are in it (from a different helm version)
> 
>     * help avoid the issue happening again?

#7653 allows you to upgrade a release that is stuck in the `FAILED` state, so it should help in both cases.

**Comment by bacongobbler** [comment]
The issue described by the OP has been solved in Helm 3.2.3, so this ticket can be closed. Please upgrade to 3.2.3 to take advantage of this feature. Thank you!

**Comment by brianharwell** [comment]
I installed helm v3.2.3 and it fixes the issue. Thanks!
